trotters/trotters.py

```diff
@@ -0,0 +1,259 @@
+        pass
+
+    async def initialize(self):
+        await self.register_casetypes()
+
+    @staticmethod
+    async def register_casetypes():
+        # register mod case
+        role_case = {
+            "name": "Role List Update",
+            "default_setting": True,
+            "image": "\N{PAGE FACING UP}",
+            "case_str": "roleupdate",
+        }
+        try:
+            await modlog.register_casetype(**role_case)
+        except RuntimeError:
+            pass
+
+    # Commands
+    @commands.group()
+    @commands.guild_only()
+    async def trotters(self, ctx: Context):
+        """Trotter's Pass settings"""
+        pass
+
+    @checks.admin_or_permissions(manage_roles=True)
+    @trotters.command(name="set")
+    async def set_role(self, ctx, role: discord.Role, enabled: bool):
+        """
+        Sets role to be addable/removable
+        """
+
+        if not role:
+            return await ctx.maybe_send_embed("No role supplied")
+
+        if not enabled:
+            if await self.config.role(role).USERS():
+                pred = MessagePredicate.yes_or_no(ctx)
+            await ctx.send(f"Found logs for role {role}, do you want to erase them?")
+            try:
+                await self.bot.wait_for("message", check=pred, timeout=30)
+            except asyncio.TimeoutError:
+                return await ctx.send("Timed out.")
+            if pred.result:
+                await self.config.role(role).USERS.set({})
+            await self.config.role(role).addable.set(False)
+            await ctx.tick()
+        else:
+            await self.config.role(role).addable.set(True)
+            await ctx.tick()
+
+    @checks.admin_or_permissions(manage_roles=True)
+    @trotters.command(name="roles")
+    async def view_roles(self, ctx):
+        """
+        Lists all roles which can be added/removed
+        """
+        enabled = list()
+        for role in ctx.guild.roles:
+            if await self.config.role(role).addable() and not role.is_default():
+                enabled.append(role.name)
+        await ctx.maybe_send_embed(f"List of addable roles: \n%s" % (", ".join(enabled)))
+
+    @commands.bot_has_permissions(manage_roles=True)
+    @trotters.command(name="add")
+    async def add_role(self, ctx: GuildContext, user_id: int, role: discord.Role, *, reason: str = "Added by command."):
+        """
+        Adds specified role to given user
+        """
+
+        if not await self.config.role(role).addable():
+            return await ctx.maybe_send_embed("Role isn't set as addable.")
+        try:
+            member = ctx.guild.get_member(user_id)
+            if role in member.roles:
+                return await ctx.maybe_send_embed("Member already has that role.")
+
+            data = await self.config.role(role).USERS()
+            if len(ctx.message.attachments):
+                attachment = ctx.message.attachments[0]
+                case = await modlog.create_case(
```

Comment:
you can probably do the attachment check and instead of code to create the case in two different spots with setting the reason or not, just update the reason string here then after the attachment check create the case with the reason set here.